stdlib/tools/swift-reflection-test/swift-reflection-test.c

```diff
@@ -198,21 +207,54 @@ static void PipeMemoryReader_freeBytes(void *reader_context, const void *bytes,
 static
 const void *PipeMemoryReader_readBytes(void *Context, swift_addr_t Address,
                                        uint64_t Size, void **outFreeContext) {
-  const PipeMemoryReader *Reader = (const PipeMemoryReader *)Context;
-  uintptr_t TargetAddress = Address;
-  size_t TargetSize = (size_t)Size;
-  DEBUG_LOG("Requesting read of %zu bytes from 0x%" PRIxPTR,
-            TargetSize, TargetAddress);
-  int WriteFD = PipeMemoryReader_getParentWriteFD(Reader);
-  write(WriteFD, REQUEST_READ_BYTES, 2);
-  write(WriteFD, &TargetAddress, sizeof(TargetAddress));
-  write(WriteFD, &TargetSize, sizeof(size_t));
-  
+  PipeMemoryReader *Reader = (PipeMemoryReader *)Context;
+
+  PipeMemoryReaderPage *Page = Reader->Pages;
+  while (Page != NULL) {
+    if (Page->BaseAddress <= Address
+	&& (Page->BaseAddress + Page->Size >= Address + Size)) {
+      break;
+    }
+    Page = Page->Next;
+  }
+
+  if (Page == NULL) {
```

Comment:
If there isn't one, round the request to a multiple of aligned pages and request that.